### PR TITLE
FEAT: Contact Us CU-3r2m492

### PR DESCRIPTION
**Pull request by drewlacob**
- add contact us endpoint in user controller (could be moved?) which
  does the following :
    - checks validity of input
    - inserts the contact into the db
    - sends an email to sales with the message provided
- change routes and jwt for this endpoint
- add endpoint to FE api requests
- change contact us page to call new api, change a few hooks and things
  to track service and industry and submit these as well
- add contacts table in db

- merge with develop to avoid merge conflict
- clean up various parts of new contact us endpoint
- fix timestamp on db
- change to correct recipient email
- tested and verified working on browser and through rest client

### CODE REVIEW

**Comment by freedom-rains** [comment]
Task linked: [CU-3r2m492 BE - create endpoint for contact form](https://app.clickup.com/t/3r2m492)

**Comment by drewlacob** [comment]
Also thinking we may want another designation for routes as these continue to grow... maybe utils for stuff like contact us and forgot password?

File: server/controllers/userControllers/user.controller.js [hunk 1]
```diff
@@ -146,7 +146,44 @@ module.exports.forgotPassword = async (req, res) => {
...
+  var sent = await sendEmail('info@freedomrains.com', req.body.email, subject, text);
+
+  if (sent)
+    res.status(200).end();
+  else
+    res.status(500).end();
+};
+
+module.exports.contactUs = async (req, res) => {
+  console.log(req.body);
```
Comment (RealDavidH) [inline server/controllers/userControllers/user.controller.js]: Get rid of console.log

File: server/controllers/userControllers/user.controller.js [hunk 2]
```diff
@@ -146,7 +146,44 @@ module.exports.forgotPassword = async (req, res) => {
...
+  //error checking
+  const checkValidate = validateInput(req.body);
+  if (checkValidate.length > 0) {
+    return res.status(400).json(checkValidate);
+  }
+  
+  console.log('inserting');
```
Comment (RealDavidH) [inline server/controllers/userControllers/user.controller.js]: Do you need this console?

File: server/controllers/userControllers/user.controller.js [hunk 3]
```diff
@@ -146,7 +146,44 @@ module.exports.forgotPassword = async (req, res) => {
...
+        phone: req.body.phone,
+        industry: req.body.industry,
+        website: req.body.website,
+        service: req.body.service
+      },
+      ['contact_id', 'first_name', 'last_name', 'email', 'phone', 'industry', 'website', 'service']
+    )
+    .catch((err) => res.status(500).json(err));
+
+    console.log('sending email');
```
Comment (RealDavidH) [inline server/controllers/userControllers/user.controller.js]: another console.log

File: server/controllers/userControllers/user.controller.js [hunk 4]
```diff
@@ -146,7 +146,44 @@ module.exports.forgotPassword = async (req, res) => {
...
+  //send email to info@freedomrains.com
```
Comment (RealDavidH) [inline server/controllers/userControllers/user.controller.js]: you can get rid of this

File: server/controllers/userControllers/user.controller.js [hunk 5]
```diff
@@ -146,7 +146,44 @@ module.exports.forgotPassword = async (req, res) => {
...
+  var subject = 'Message from ' + req.body.email;
+  //change to info@freedomrains.com as receiver of email
```
Comment (RealDavidH) [inline server/controllers/userControllers/user.controller.js]: can also get rid of this.

File: server/controllers/userControllers/user.controller.js [hunk 6]
```diff
@@ -146,7 +146,44 @@ module.exports.forgotPassword = async (req, res) => {
...
+  var sent = await sendEmail(req.body.email, 'bishop7rook@gmail.com', subject, req.body.message);
```
Comment (RealDavidH) [inline server/controllers/userControllers/user.controller.js]: Get rid of bishop email. you could also turn this into a .then/.catch

sendEmail(req.body.email, 'bishop7rook@gmail.com', subject, req.body.message)
            .then(...)
            .catch(...)

File: server/routes/userRoutes/user.route.js [hunk 7]
```diff
@@ -8,4 +8,6 @@ module.exports = (app) => {
   app.put('/users/update/:id', updateUser); // updates a user. Requires (ID)
   app.delete('/users/delete/:id', deleteUser); // deletes a user. Requires (ID)
   app.post('/users/forgotPassword', forgotPassword); // sets temp password emails it
-};
\ No newline at end of file
+  app.post('/users/contactUs', contactUs); // creates contact in db, emails sales with message
```
Comment (RealDavidH) [inline server/routes/userRoutes/user.route.js:11]: This is outside the routes

File: server/controllers/userControllers/user.controller.js [hunk 8]
```diff
@@ -146,7 +146,40 @@ module.exports.forgotPassword = async (req, res) => {
   await knex.schema.raw(`UPDATE users SET password = '${hashedTempPass}' WHERE username = '${req.body.email}';`);
```
Comment (RealDavidH) [inline server/controllers/userControllers/user.controller.js:146]: I think in our users table it's "email", not "username"

**Review by RealDavidH — CHANGES_REQUESTED** [review]

File: server/controllers/userControllers/user.controller.js (hunk 8 above)
Comment (drewlacob) [reply server/controllers/userControllers/user.controller.js:146]: Good catch. For most of the entries it appears there is both username and email which are the email of the user. Will change to email tho as that will be 100% of the time.

File: server/controllers/userControllers/user.controller.js [hunk 9]
```diff
@@ -146,7 +146,40 @@ module.exports.forgotPassword = async (req, res) => {
...
+        website: req.body.website,
+        service: req.body.service
+      },
+      ['contact_id', 'first_name', 'last_name', 'email', 'phone', 'industry', 'website', 'service']
+    )
+    .catch((err) => res.status(500).json(err));
+
+  //send email with message to info@freedomrains.com
+  var subject = 'Message from ' + req.body.email;
+  var sent = await sendEmail(req.body.email, 'info@freedomrains.com', subject, req.body.message);
```
Comment (freedom-rains) [inline server/controllers/userControllers/user.controller.js:182]: The email being sent should include the message (which you have) but also the user’s first and last name. As well as the other info from the contact form.
You may have to create a new variable containing all that info and pass that variable as your text

**Review by freedom-rains — CHANGES_REQUESTED** [review]

**Review by freedom-rains — APPROVED** [review]

**Review by RealDavidH — APPROVED** [review]

**Pull request merged**